tigerpath/majors_and_certificates/scripts/verifier.py

```diff
@@ -316,6 +318,65 @@ def _format_courses_output(courses):
                 output[i][j]["settled"] = course["settled"]
     return output
 
+def _parse_year_code(code, year):
+    """
+    Returns whether `year` falls in the range specified by `code`
+    """
+    if isinstance(code, int):  # explicitly specified year as an int
+        return year == code
+    if not code or code.lower() == "default":  # empty indicates default case
+        return True
+    code = code.replace(" ", "")  # strip it of spaces for processing
+    if code.startswith("<="):
+        return year <= int(code[2:])
+    elif code.startswith(">="):
+        return year >= int(code[2:])
+    elif code.startswith("<"):
+        return year < int(code[1:])
+    elif code.startswith(">"):
+        return year > int(code[1:])
+    elif code.startswith("=="):
+        return year == int(code[2:])
+    elif code.startswith("!="):
+        return year != int(code[2:])
+    elif "-" in code:  # range of years (inclusive), such as "2018-2020"
+        fr, to, *_ = code.split("-")
+        return year >= int(fr) and year <= int(to)
+    else:  # just the year is the same as ==
+        return year == int(code)
+
+def _init_year_switch(req, year):
+    """
+    Checks for any year_switch primitives and selects the right subrequirements.
+
+    Any requirement that contains a year_switch is overridden by the first
+    subrequirement of that year_switch whose year code matches the user's
+    class year.
+    Fields in req are overridden by any explicitly listed fields of the
+    overriding subrequirement, but any fields not specified by the
+    subrequirement remain as is (except the year_switch, which is removed).
+
+    If no year code matches, the requirement is unchanged and the year_switch
+    is just removed (that is, the year_switch is ignored and has no effect).
+    """
+    if "year_switch" in req:
+        newreq = {}
+        for subreq in req["year_switch"]:
+            code = subreq.get("year_code", None)  # year_code set to default
+            if _parse_year_code(code, year):
```

Comment:
This is the only other place where this function appears

```suggestion
            if _year_matches_code(year, code):
```